### PR TITLE
Connor/waittimes dragging

### PR DESCRIPTION
**Pull request by connorreinhold**
**Overview**
Makes dragging functionality of the wait times component much cleaner.
This is the exact code of the last PR that was reverted, except gone back one commit and updated with only linting the java files

**Changes Made**
Messed with the onTouch method of the waittimes component. Low-level calculations of direction of finger motion were used to either allow interaction with the wait times component or prevent interaction.

Extended WaitTimesChart in order to override performClick() (this was necessary)

**Test Coverage**
Tested on phone and on simulator. Dragging works a lot better than before and clickable functionality was not harmed.

**Next Steps**
There's this slight issue where sometimes if you drag horizontally and then drag vertically the vertical swipe is registered as a horizontal swipe. This is because the location the component gives on the down pressed of onTouch is inaccurate -- normally is 4/5 pixels off in a random direction and it doesn't make any sense. The issue is not noticeable unless you are trying to break the component, so I vote we ignore it

**Related PRs or Issues**
Addresses issue #163

### CODE REVIEW

File: app/src/main/java/com/cornellappdev/android/eatery/model/DiningHallModel.java [hunk 1]
```diff
@@ -36,7 +36,7 @@ public static DiningHallModel fromJSON(Context context, boolean hardcoded, JSONO
     }
 
     public static DiningHallModel fromEatery(Context context, boolean hardcoded,
-            AllEateriesQuery.Eatery eatery) {
+                                             AllEateriesQuery.Eatery eatery) {
```
Comment (YKo20010) [inline app/src/main/java/com/cornellappdev/android/eatery/model/DiningHallModel.java]: can this fit on one line?

File: app/src/main/java/com/cornellappdev/android/eatery/model/CollegeTownModel.java [hunk 2]
```diff
@@ -37,7 +36,7 @@ public CollegeTownModel() {
     }
 
     public static CollegeTownModel fromEatery(Context context,
-            AllCtEateriesQuery.CollegetownEatery ctEatery) {
+                                              AllCtEateriesQuery.CollegetownEatery ctEatery) {
```
Comment (YKo20010) [inline app/src/main/java/com/cornellappdev/android/eatery/model/CollegeTownModel.java]: can this fit on one line?

File: app/src/main/java/com/cornellappdev/android/eatery/model/CafeModel.java [hunk 3]
```diff
@@ -64,7 +64,7 @@ public static CafeModel fromJSONObject(Context context, boolean hardcoded, JSONO
     }
 
     public static CafeModel fromEatery(Context context, boolean hardcoded,
-            AllEateriesQuery.Eatery cafe) {
+                                       AllEateriesQuery.Eatery cafe) {
```
Comment (YKo20010) [inline app/src/main/java/com/cornellappdev/android/eatery/model/CafeModel.java]: can this fit on one line?

File: app/src/main/java/com/cornellappdev/android/eatery/model/DiningHallModel.java (hunk 1 above)
Comment (connorreinhold) [reply app/src/main/java/com/cornellappdev/android/eatery/model/DiningHallModel.java]: No its over the limit

**Pull request merged**